pkg/controller/plan/migration.go

```diff
@@ -916,7 +943,124 @@ func (r *Migration) resetPrecopyTasks(vm *plan.VMStatus, step *plan.Step) {
 	}
 }
 
-//
+func (r *Migration) createVolumes(vm ref.Ref) (err error) {
+	if r.vmMap == nil {
+		r.vmMap, err = r.kubevirt.VirtualMachineMap()
+		if err != nil {
+			return
+		}
+	}
+	ovirtVm := &ovirt.Workload{}
+	err = r.Source.Inventory.Find(ovirtVm, vm)
+	if err != nil {
+		return
+	}
+	url, err := url.Parse(r.Source.Provider.Spec.URL)
+	if err != nil {
+		return
+	}
+
+	storageName := &r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
+	for _, da := range ovirtVm.DiskAttachments {
+		populatorCr := v1beta1.OvirtImageIOPopulator{
+			ObjectMeta: meta.ObjectMeta{
+				Name:      da.DiskAttachment.ID,
+				Namespace: r.Plan.Spec.TargetNamespace,
+			},
+			Spec: v1beta1.OvirtImageIOPopulatorSpec{
+				EngineURL:        fmt.Sprintf("https://%s", url.Host),
+				EngineSecretName: r.Source.Secret.Name,
+				DiskID:           da.Disk.ID,
+			},
+		}
+
+		err = r.Client.Create(context.Background(), &populatorCr, &client.CreateOptions{})
+		if err != nil {
+			return
+		}
+		apiGroup := "forklift.konveyor.io"
+
+		pvc := &unstructured.Unstructured{
```

Comment:
why it's unstructured and not core.PersistentVolumeClaim?

Reply:
Because we need the dataSourceRef field which doesn't exist in the struct on the version we use, if we upgrade the client then we should switch to the typed version, I thought I had a TODO for this but it was probably in the other version

Reply:
@bennyz as it's a repeating question, it is worth to document it..

Reply:
added

Reply:
once we will bump the code dep, we can drop this (in https://github.com/kubev2v/forklift/pull/64)

Reply:
> once we will bump the code dep, we can drop this (in #64)

that's fine but it means we won't support OCP 4.10, right?

Reply:
basing on https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/ it's since k8s 1.22, OCP 4.9 uses k8s 1.22. OCP 4.10 uses 1.23.

Reply:
> basing on https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/ it's since k8s 1.22, OCP 4.9 uses k8s 1.22. OCP 4.10 uses 1.23.

`In v1.22 if the AnyVolumeDataSource feature gate is enabled, ..` and ask @bennyz about his experience in enabling it on such cluster..
anyway, now that OCP 4.12 is out, we don't care about 4.10 and it's enabled by default on OCP 4.11 so that's alright